Sources/StreamChat/Query/Filter+ChatChannel.swift

```diff
@@ -13,3 +13,194 @@ extension ChatChannel {
+            guard
+                let filters = value as? [Filter<Scope>]
+            else {
+                return nil
+            }
+            return NSCompoundPredicate(
+                orPredicateWithSubpredicates: filters.compactMap(\.predicate)
+            )
+        case .nor where value is [Filter<Scope>]:
+            guard
+                let filters = value as? [Filter<Scope>]
+            else {
+                return nil
+            }
+            return NSCompoundPredicate(
+                notPredicateWithSubpredicate: NSCompoundPredicate(
+                    orPredicateWithSubpredicates: filters.compactMap(\.predicate)
+                )
+            )
+        default:
+            log.debug("Unhandled operator \(`operator`) and filterValue \(value)")
+            return nil
+        }
+    }
+
+    private func comparingPredicate(
+        _ operator: FilterOperator
+    ) -> NSPredicate? {
+        guard
+            key != nil,
+            let keyPathValueProvider = keyPathValueProvider
+        else {
+            return nil
+        }
+
+        switch `operator` {
+        case .equal:
+            return NSPredicate(
+                format: "%@ == %K",
+                argumentArray: [value, keyPathValueProvider()]
+            )
+        case .notEqual:
+            return NSPredicate(
+                format: "%@ != %K",
+                argumentArray: [value, keyPathValueProvider()]
+            )
+        case .greater:
+            return NSPredicate(
+                format: "%K > %@",
+                argumentArray: [keyPathValueProvider(), value]
+            )
+        case .greaterOrEqual:
+            return NSPredicate(
+                format: "%K >= %@",
+                argumentArray: [keyPathValueProvider(), value]
+            )
+        case .less:
+            return NSPredicate(
+                format: "%K < %@",
+                argumentArray: [keyPathValueProvider(), value]
+            )
+        case .lessOrEqual:
+            return NSPredicate(
+                format: "%K <= %@",
+                argumentArray: [keyPathValueProvider(), value]
+            )
+        default:
+            log.debug("Unhandled operator \(`operator`) and filterValue \(value)")
+            return nil
+        }
+    }
+
+    private func collectionPredicate(
+        _ operator: FilterOperator
+    ) -> NSPredicate? {
+        guard
+            key != nil,
+            let keyPathValueProvider = keyPathValueProvider
+        else {
+            return nil
+        }
+
+        switch `operator` {
+        case .in where value is [FilterValue]:
+            guard
+                let filterArray = (value as? [FilterValue])
+            else {
+                return nil
+            }
+            return NSCompoundPredicate(
+                andPredicateWithSubpredicates: filterArray.map { subValue in
+                    NSPredicate(
+                        format: "%@ IN %K",
```

Comment:
Oh wait, was the issue in my implementation that we were using %@ instead of %K? 🤦

Reply:
Exactly ;) %K does the trick

Reply:
> This might not work as %K is usually not a top-level property

I'm not sure what you mean by this. `%K` represents keypaths and doesn't seem to have any limitations on what level this keypath is

Reply:
I might have spent hours trying other things, and that was it. Great 😂

Reply:
It was literally after had spent hours on documentation and trials and couldn't figure out why it doesn't work, that i read this one 
> If you want to specify a dynamic property name, use %K in the format string, as shown in the following example.
[Apple Docs](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Predicates/Articles/pSyntax.html)